### ISSUE TITLE
invariant failure

### ISSUE THREAD

**Issue by fommil** [scalameta/scalafmt#1123]
scalafmt 1.4.0 fails to parse this file https://github.com/ensime/ensime-server/blob/3.0/json/src/main/scala/spray/json/JsPrinter.scala

```
[error] Error in /home/fommil/Projects/ensime-server/json/src/main/scala/spray/json/JsPrinter.scala: org.scalameta.invariants.InvariantFailedException: invariant failed:
[error] when verifying roughIndex.>=(0).&&(org.scalameta.`package`.debug(XtensionTokenIndex.this.token))
[error] found that roughIndex.>=(0) is false
[error] where XtensionTokenIndex = scala.meta.internal.parsers.ScalametaParser$XtensionTokenIndex@13205550
[error] where XtensionTokenIndex.this.token = (
[error] where roughIndex = -1
```

Admittedly, it's probably doing something funky.

**Comment by fommil** [comment]
I can't reproduce anymore, so closing as a dud.

**Comment by rpless** [comment]
I'm seeing this error on scalafmt 1.5.1 when trying format `Finch`. I get:
```
[error] Error in /home/rpless/projects/finch/core/src/main/scala/io/finch/Endpoint.scala: org.scalameta.invariants.InvariantFailedException: invariant failed:
[error] when verifying roughIndex.>=(0).&&(org.scalameta.`package`.debug(XtensionTokenIndex.this.token))
[error] found that roughIndex.>=(0) is false
[error] where XtensionTokenIndex = scala.meta.internal.parsers.ScalametaParser$XtensionTokenIndex@6d087aab
[error] where XtensionTokenIndex.this.token = (
[error] where roughIndex = -1
```
from this [file](https://github.com/finagle/finch/blob/f72d0fcdf1675eabc87ad40e27aa29a6c94b95cc/core/src/main/scala/io/finch/Endpoint.scala).
I doesn't seem to stop my build and scalafmt appears to reformat that file correctly anyway.

**Comment by olafurpg** [comment]
Thanks for reporting! I am unable to reproduce on the file https://github.com/finagle/finch/blob/f72d0fcdf1675eabc87ad40e27aa29a6c94b95cc/core/src/main/scala/io/finch/Endpoint.scala (at that exact commit). I tried

```
git clone ...
cd finch
git checkout f72d0fcdf1675eabc87ad40e27aa29a6c94b95cc
rm .scalafmt.conf // to ignore verticalMultiline setting
coursier launch com.geirsson:scalafmt-cli_2.12:1.5.1  -- --stdout core/src/main/scala/io/finch/Endpoint.scala
```

Can you also try with the latest release candidate? I think it's v1.6.0-RC4.

**Comment by rpless** [comment]
Hrm, that commit is after we applied the new formatting, so maybe you want to try from [the previous commit](https://github.com/finagle/finch/pull/997/commits/de1feb78a77963598bac82056afb4e7e6921895c).

Either way I did a little more digging:
- Per your suggestion, I tried `1.6.0-RC-4` and the issue no longer appears.
- I tried applying `scalafmt` from the commit I mentioned on another computer and it didn't happen. The difference that stands out to me is the operating systems are different. The computer that the error appears on runs Ubuntu 18.04 and the computer it does not appear on is a Mac.

It seems like this issue is fixed in 1.6.0.

As an aside, I'm curious why you deleted the existing `.scalafmt.conf`? Should we not be using the `verticalMultiline` setting?

**Comment by olafurpg** [comment]
> It seems like this issue is fixed in 1.6.0.
Awesome, will close this as fixed then, yet more more reason to get around to finish that v1.6 release 😅 

> The difference that stands out to me is the operating systems are different. The computer that the error appears on runs Ubuntu 18.04 and the computer it does not appear on is a Mac.

Hmm, I've seen issues with windows vs. non-windows but never ubuntu vs. macos 🤔 

> As an aside, I'm curious why you deleted the existing .scalafmt.conf? Should we not be using the verticalMultiline setting?

The setting doesn't exist in v1.5.1 so I was unable to run scalafmt because of it. No other reason :)

**Comment by fommil** [comment]
Can you please create a fresh ticket with your reproduction?

**Comment by peter-empen** [comment]
Same issue here. Really difficult to reproduce. It may or may not occur on different Ubuntu machines.
Not sure but JVM settings or the executed SBT task sequence may have an impact.

**Comment by peter-empen** [comment]
The reason you have closed this issue is that 1.6.0 includes a fix, right?

**Comment by fommil** [comment]
You're reporting a different issue, please open a new one.
